File: ClockApp/Clock.cs
namespace ClockApp;

public class Clock
{
    private readonly Counter _seconds;
    private readonly Counter _minutes;
    private readonly Counter _hours;

    public Clock()
    {
        _hours = new Counter("Hours");
        _minutes = new Counter("Minutes");
        _seconds = new Counter("Seconds");
    }

    public void Tick()
    {
        _seconds.Increment();
        if (_seconds.Tick != 60)
        {
            return;
        }
        _seconds.Reset();
        _minutes.Increment();
        if (_minutes.Tick != 60)
        {
            return;
        }
        _minutes.Reset();
        _hours.Increment();
        if (_hours.Tick != 24)
        {
            return;
        }
        _hours.Reset();
    }

    public void Reset()
    {
        _seconds.Reset();
        _minutes.Reset();
        _hours.Reset();
    }

    public string Time => $"{_hours.Tick:D2}:{_minutes.Tick:D2}:{_seconds.Tick:D2}";
}

File: Clock/clock.py
from counter import Counter
class Clock:
    def __init__(self):
        self._seconds = Counter("seconds")
        self._minutes = Counter("minutes")
        self._hours = Counter("hours")

    def tick(self):
        self._seconds.increment()
        if self._seconds.ticks != 60:
            return

        self._seconds.reset()
        self._minutes.increment()
        if self._minutes.ticks != 60:
            return

        self._minutes.reset()
        self._hours.increment()
        if self._hours.ticks != 24:
            return

        self._hours.reset()

    def reset(self):
        self._seconds.reset()
        self._minutes.reset()
        self._hours.reset()

    @property
    def time(self):
        return (
            f"{self._hours.ticks:02}:{self._minutes.ticks:02}:{self._seconds.ticks:02}"
        )